src/sentry/reprocessing2.py

```diff
@@ -303,20 +303,25 @@ def _get_info_reprocessed_key(group_id):
     return f"re2:info:{group_id}"
 
 
-def mark_event_reprocessed(data):
+def mark_event_reprocessed(data=None, group_id=None, project_id=None):
     """
     This function is supposed to be unconditionally called when an event has
     finished reprocessing, regardless of whether it has been saved or not.
     """
-    group_id = _get_original_issue_id(data)
-    if group_id is None:
-        return
+    if data is not None:
+        assert group_id is None
+        assert project_id is None
+        group_id = _get_original_issue_id(data)
+        if group_id is None:
+            return
+
+        project_id = data["project"]
 
-    key = _get_sync_counter_key(_get_original_issue_id(data))
+    key = _get_sync_counter_key(group_id)
```

Comment:
Outside the scope of this PR, but why do we require a redis counter here, instead of scheduling `finish_reprocessing` after the final batch?

Reply:
this counter drives progressbar in the UI. I want to only advance the progressbar when the event has made it through process_event, symbolicate_event and save_event which can take a few *minutes* when native stacktrace processing is involved and caches are cold

this is also why I can't call mark_event_reprocessed in the finally-block, because it's way too soon to do that. in the case where I put events into remaining_events, they are "basically instantly" done so I can call it immediately.